src/strategies/spidStrategy.ts

```diff
@@ -129,4 +134,29 @@ const spidStrategy = async (
   );
 };
 
+/**
+ * Reads dates information in x509 certificate and logs remaining time to its expiration date.
+ * @param samlCert x509 certificate as string
+ */
+function logSamlCertExpiration(samlCert: string): void {
+  try {
+    const tempDir = fs.mkdtempSync("saml-cert");
+    fs.writeFileSync(`${tempDir}/samlCert.pem`, samlCert);
+    const out = child_process.execSync(
+      `openssl x509 -in ${tempDir}/samlCert.pem -noout -dates`
+    );
+    fs.unlinkSync(`${tempDir}/samlCert.pem`);
+    const matchResult = /notAfter=(.*)$/m.exec(out.toString());
+    if (matchResult) {
+      const expireAt = new Date(matchResult[1]);
+      const timeDiff = expireAt.valueOf() - Date.now();
+      log.info("samlCert expire in %s seconds", (timeDiff / 1000).toFixed(0));
```

Comment:
you can use [formatDistance](https://date-fns.org/) to make the time distance human readable

also it could be useful to log to use `log.warn` if the distance is less than 60 days

Reply:
fixed in 42c0a16